data/helpers.d/setting

```diff
@@ -158,48 +158,155 @@ ynh_webpath_register () {
 
 # Create a new permission for the app
 #
-# example: ynh_permission_create --permission=admin --url=/admin --allowed="alice bob"
-#
-# usage: ynh_permission_create --permission "permission" [--url=url] [--allowed="group1 group2"]
-# | arg: -p, --permission=  - the name for the permission (by default a permission named "main" already exist)
-# | arg: -u, --url=         - (optional) URL for which access will be allowed/forbidden
-# | arg: -a, --allowed=     - (optional) A list of group/user to allow for the permission
-#
-# If provided, 'url' is assumed to be relative to the app domain/path if they
+# example 1: ynh_permission_create --permission=admin --url=/admin --additional_urls=domain.tld/admin /superadmin --allowed=alice bob \
```

Comment:
Thanks for the comment header.
The purpose and usages of this helper is clearer now.

Just a few questions though
Why here --additional_urls=**domain.tld**/admin if the path is relative ?
You explain later that the path is relative, and yet you add again the domain. Is there a reason to add the domain before the path, and isn't there a risk that the domain is concatenated to the real domain ?

Reply:
> Why here --additional_urls=domain.tld/admin if the path is relative ?

No it's full path.

Maybe it's not really clear. But the idea is here to give the possibility to have multiple path and also multiple domain for the same app. An app have a "main" domain and a "main" path which is set in the settings. You can define a permission with a path relative to this main path. In this case it start with `/`. If you need more flexibility you need to use a full path with the domain specified.

Reply:
I'm not sure to have fully get your answer.

Line 198, the comment says clearly that the path is "assumed to be relative"
And in the example below, you show a domain.tld.

So, it's relative only if it starts by a /, otherwise it's not ?

And regex can be relative as well ?
Line 206, the comment let think that a relative regex will be concatenated to the domain anyway. what would happen with a domain with a `-` in the middle ?
Those are known to failed the regex into the SSO.

Reply:
> Line 198, the comment says clearly that the path is "assumed to be relative"

Only if it start with `/` cf line 199

> And in the example below, you show a domain.tld.
> 
> So, it's relative only if it starts by a /, otherwise it's not ?

Yes

> And regex can be relative as well ?

Yes (line 206)

> Line 206, the comment let think that a relative regex will be concatenated to the domain anyway. what would happen with a domain with a - in the middle ?

Oh yes, it's not specified. The regex need to be a PCRE regex not lua regex. Note that the '.' in the domain is automatically replaced by `\.` when it's a relative regex.

Reply:
If a regex is relative, then is concatenated to the domain ? Really important because it currently does not work and grab all other domains matching the regex.

> what would happen with a domain with a - in the middle ?

Not a `.`, a `-`. That currently fail the regex.

The comment you've added says `PCRE (lua)`. Did you mean `PCRE (not lua)` ?

Reply:
> If a regex is relative, then is concatenated to the domain ? Really important because it currently does not work and grab all other domains matching the regex.

Yes it's concatenated with the domain and the app path.

> > what would happen with a domain with a - in the middle ?

As I know `-` is not considered a special shar with PCRE regex.

> The comment you've added says `PCRE (lua)`. Did you mean `PCRE (not lua)` ?

Oh yes, my bad.

Reply:
Maybe it was just a problem with the lua "regex". Hope so...
`-` is used to separate a fields of value.